test/index.js

```diff
@@ -15,32 +17,413 @@ test.describe('input fields', function(){
+     const newTitle = driver.findElement({name: 'todo-title'});
+     return newTitle.getText();
    }).then((text)=>{
-     assert.equal(text, 'edit this is a title')
-   })
- })
-})
+     assert.equal(text, 'edit this is a title');
+   });
+ });
+
+ test.it('should filter todos based on a search string live', ()=>{
+   const title      = driver.findElement({name: 'title'});
+   const task       = driver.findElement({name: 'todo'});
+   const saveBtn    = driver.findElement({name: 'saveBtn'});
+   const textSearch = driver.findElement({name: 'text-search'});
+   title.sendKeys('Title');
+   task.sendKeys('this is a todo');
+   saveBtn.click();
+   const card = driver.findElement({className: 'todo-card'});
+   textSearch.sendKeys('t');
+   card.isDisplayed().then((displayed)=>{
+      assert.equal(displayed,true);
+   });
+   textSearch.sendKeys('z');
+   card.isDisplayed().then((displayed)=>{
+      assert.equal(displayed,false);
+   });
+  });
+
+  test.it('should move up the quality array when the up vote button is pressed and stop at critical',()=>{
+    const title   = driver.findElement({name: 'title'});
+    const task    = driver.findElement({name: 'todo'});
+    const saveBtn = driver.findElement({name:'saveBtn'});
+    title.sendKeys('this is a title');
+    task.sendKeys('this is a todo');
+    saveBtn.click();
+    const upVote    = driver.findElement({className: 'up-vote'});
+    const downVote  = driver.findElement({className: 'down-vote'});
+    const quality   = driver.findElement({className: 'quality'});
+    const qualityArrayTest = ['None','Low','Normal','High','Critical'];
+
+    for (let i = 0;i<5;i++){
+      quality.getText().then((q)=>{
+        assert.equal(q,qualityArrayTest[i]);
+        upVote.click();
+      });
+    }
+
+    upVote.click();
+    quality.getText().then((q)=>{
+      assert.equal(q,'Critical');
+    });
+  });
+
+  test.it('should move down the quality array when the down vote button is pressed and stop at none',()=>{
+    const title   = driver.findElement({name: 'title'});
+    const task    = driver.findElement({name: 'todo'});
+    const saveBtn = driver.findElement({name:'saveBtn'});
+    title.sendKeys('this is a title');
+    task.sendKeys('this is a todo');
+    saveBtn.click();
+    const upVote    = driver.findElement({className: 'up-vote'});
+    const downVote  = driver.findElement({className: 'down-vote'});
+    const quality   = driver.findElement({className: 'quality'});
+    const qualityArrayTest = ['Critical','High','Normal','Low','None'];
+
+    for (let i = 0; i<5;i++){
+      upVote.click();
+    }
+
+    for (let i = 0;i<5;i++){
+      quality.getText().then((q)=>{
+        assert.equal(q,qualityArrayTest[i]);
+        downVote.click();
+      });
+    }
+    downVote.click();
+    quality.getText().then((q)=>{
+      assert.equal(q,'None');
+    });
+  });
+  test.it('should only show todos with none quality when the none filter button is pressed',()=>{
```

Comment:
these tests are a little verbose. I would love some suggestions on how to trim them down